src/main/java/com/yahoo/validatar/report/email/EmailFormatter.java

```diff
@@ -0,0 +1,247 @@
+                    passCount++;
+                }
+            }
+            this.testPassed = passCount;
+            this.testTotal = testSuite.tests.size();
+        }
+
+        /**
+         * @return true if all queries and tests passed
+         */
+        protected boolean allPassed() {
+            return queryPassed == queryTotal && testPassed == testTotal;
+        }
+    }
+
+    private static final String EMAIL_FORMATTER = "email";
+
+    private static final String RECIPIENT_EMAILS = "recipient-emails";
+    private static final String SENDER_NAME = "sender-name";
+    private static final String FROM_EMAIL = "from-email";
+    private static final String REPLY_TO = "reply-to";
+    private static final String SMTP_HOST = "smtp-host";
+    private static final String SMTP_PORT = "smtp-port";
+
+    /**
+     * Option parser for this class. Caller should provide
+     * the following arguments:
+     * <ul>
+     *     <li>--recipient-emails | list of emails to send reports</li>
+     *     <li>--sender-name | name of the report email sender (default 'Validatar'</li>
+     *     <li>--from-email | email to show as the sender</li>
+     *     <li>--reply-to | email to which replies are sent</li>
+     *     <li>--smtp-host | SMTP server host</li>
+     *     <li>--smtp-port | SMTP server port</li>
+     * </ul>
+     */
+    private static final OptionParser PARSER = new OptionParser() {
+        {
+            accepts(RECIPIENT_EMAILS, "Comma-separated list of emails to send reports")
+                    .withRequiredArg()
+                    .describedAs("Report recipients' emails");
+            accepts(SENDER_NAME, "Name of sender displayed to report recipients")
+                    .withRequiredArg()
+                    .defaultsTo("Validatar");
+            accepts(FROM_EMAIL, "Email shown to recipients as 'from'")
+                    .withRequiredArg();
+            accepts(REPLY_TO, "Email to which replies will be sent")
+                    .withRequiredArg();
+            accepts(SMTP_HOST, "Email SMTP host name")
+                    .withRequiredArg();
+            accepts(SMTP_PORT, "Email SMTP port")
+                    .withRequiredArg();
+            allowsUnrecognizedOptions();
+        }
+    };
+
+    /**
+     * List of recipient emails for Validatar reports.
+     */
+    private List<String> recipientEmails;
+    /**
+     * Sender name to display to report recipients.
+     */
+    private String senderName;
+    /**
+     * From email to display to report recipients.
+     */
+    private String fromEmail;
+    /**
+     * Email to which report email replies are sent.
+     */
+    private String replyTo;
+    /**
+     * SMTP server host.
+     */
+    private String smtpHost;
+    /**
+     * SMTP server port.
+     */
+    private int smtpPort;
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public boolean setup(String[] arguments) {
+        OptionSet options = PARSER.parse(arguments);
```

Comment:
Might have to catch this once you add required args without defaults and return false